### PR TITLE
feat(evpn): enable tls on godpu

### PR DESCRIPTION
**Pull request by atulpatel261194**

### CODE REVIEW

File: cmd/storage/frontend/nvme_controller.go [hunk 1]
```diff
@@ -65,6 +67,7 @@ func newCreateNvmeControllerTCPCommand() *cobra.Command {
 	cmd.Flags().StringVar(&subsystem, "subsystem", "", "subsystem name to attach the controller to")
 	cmd.Flags().IPVar(&ip, "ip", nil, "ip address of the created controller")
 	cmd.Flags().Uint16Var(&port, "port", 0, "port of the created controller")
+	cmd.Flags().StringVar(&tlsFiles, "tls", "", "TLS files in client_cert:client_key:ca_cert format.")
```
Comment (artek-koltun) [inline cmd/storage/frontend/nvme_controller.go:70]: this flag will  be used by all cmds, right?
Would it make sense to define tls cmd line arg and maybe some others on the top-level? like it is done [here](https://github.com/opiproject/godpu/blob/e4af793befd97271bc33ed40c404de07c300bf8c/cmd/storage/storage.go#L30-L31)

File: grpc/client.go [hunk 2]
```diff
@@ -44,13 +46,36 @@ func NewWithDialler(address string, d Dialler) (Connector, error) {
...
 	}, nil
 }
 
 // NewConn creates a new gRPC connection
 func (c clientImpl) NewConn() (grpc.ClientConnInterface, Closer, error) {
+	log.Println("files", c.tlsfile)
+	if c.tlsfile != "" {
+		config, err := ParseTLSFiles(c.tlsfile)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to parse TLS files: %w", err)
```
Comment (artek-koltun) [inline grpc/client.go:61]: Does a client need to have access to the underlying error (%w)?

**Review by artek-koltun — COMMENTED** [review]
one of the tests is failing

File: grpc/client.go (hunk 2 above)
Comment (atulpatel261194) [reply grpc/client.go:61]: client is responsible for providing the certificate, then i believe it is ok to provide the underlying error .

File: cmd/cmd.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			os.Exit(1)
+		},
+	}
+	c.AddCommand(inventory.NewInventoryCommand())
+	c.AddCommand(ipsec.NewIPSecCommand())
+	c.AddCommand(storage.NewStorageCommand())
+	c.AddCommand(network.NewEvpnCommand())
+
+	flags := c.PersistentFlags()
+	flags.String(AddrCmdLineArg, "localhost:50151", "address of OPI gRPC server")
```
Comment (artek-koltun) [inline cmd/cmd.go:47]: addr is probably required here, right? Should we mark it with `MarkFlagRequired`?

File: cmd/ipsec/ipsec-stats.go [hunk 4]
```diff
@@ -12,23 +12,27 @@ import (
 	"github.com/spf13/cobra"
 )
 
+// AddrCmdLineArg cmdline arg name for address
+const AddrCmdLineArg = "addr"
+
+// TLSFiles cmdline arg name for tls files
+const TLSFiles = "tlsfiles"
```
Comment (artek-koltun) [inline cmd/ipsec/ipsec-stats.go:19]: is it possible to move these addr and tls constants in some common location in order not to repeat it again and again?

File: grpc/client.go (hunk 2 above)
Comment (artek-koltun) [reply grpc/client.go:61]: Usually %w is used in order a client could extract the original error here and perform recovery based on that error

File: cmd/cmd.go (hunk 3 above)
Comment (atulpatel261194) [reply cmd/cmd.go]: it is not a required flag.

Comment (artek-koltun) [reply cmd/cmd.go]: true. My oversight

File: cmd/network/evpn-utils.go [hunk 5]
```diff
@@ -52,7 +53,7 @@ func PrintBP(bp *pb.BridgePort) {
 	log.Println("name:", bp.GetName())
 	log.Println("status:", bp.GetStatus().GetOperStatus().String())
 	log.Println("ptype:", bp.GetSpec().GetPtype())
-	log.Println("mac:", bp.GetSpec().GetMacAddress())
+	log.Println("mac:", net.HardwareAddr(bp.GetSpec().GetMacAddress()).String())
```
Comment (artek-koltun) [inline cmd/network/evpn-utils.go:56]: are changes in this file related to the PR goal? If they are not TLS related, they have to be a part of a separate PR

Comment (atulpatel261194) [reply cmd/network/evpn-utils.go]: reverted back to previous commit.

**Review by artek-koltun — APPROVED** [review]
Looks good!

Created 2 issue for future improvements: https://github.com/opiproject/godpu/issues/449 https://github.com/opiproject/godpu/issues/450
